libres/lib/analysis/ies/ies_data.cpp

```diff
@@ -40,18 +41,20 @@ struct ies::data::data_struct {
         config; // This I don't understand but I assume I include data from the ies_config_type defined in ies_config.cpp
 };
 
-ies::data::data_type *ies::data::alloc(bool ies_mode) {
+ies::data::data_type *ies::data::alloc(int ens_size, bool ies_mode) {
     ies::data::data_type *data = new ies::data::data_type();
     data->iteration_nr = 0;
     data->state_size = 0;
+    data->ens_size = ens_size;
     data->ens_mask = NULL;
     data->obs_mask0 = NULL;
     data->obs_mask = NULL;
-    data->W = NULL;
     data->A0 = NULL;
     data->E = NULL;
     data->converged = false;
     data->config = ies::config::alloc(ies_mode);
+    data->W = matrix_alloc(data->ens_size, data->ens_size);
```

Comment:
This PR looks good.
I think it makes sense to allocate `W` here and it would be nice if the final commit message mentions it.

Reply:
Well spotted - that minor change was actually of some significance; I will try to pull it out in a separate commit.